apps/server/src/modules/h5p-editor/controller/dto/ajax/post.body.params.ts

```diff
@@ -0,0 +1,72 @@
+import { BadRequestException, Injectable, PipeTransform, ValidationPipe } from '@nestjs/common';
+import { AnyFilesInterceptor } from '@nestjs/platform-express';
+import { ApiProperty } from '@nestjs/swagger';
+import { plainToClass } from 'class-transformer';
+import { IsArray, IsMongoId, IsOptional, IsString, validate } from 'class-validator';
+
+class LibrariesBodyParams {
+	@ApiProperty()
+	@IsArray()
+	@IsString({ each: true })
+	libraries!: string[];
+}
+
+class ContentBodyParams {
+	@ApiProperty()
+	@IsMongoId()
+	contentId!: string;
+
+	@ApiProperty()
+	@IsString()
+	@IsOptional()
+	field!: string; // Todo: Reason
+}
+
+class LibraryParametersBodyParams {
+	@ApiProperty()
+	@IsString()
+	libraryParameters!: string;
+}
+
+export type AjaxPostBodyParams = LibrariesBodyParams | ContentBodyParams | LibraryParametersBodyParams | undefined;
+
+@Injectable()
+export class AjaxPostBodyParamsTransformPipe implements PipeTransform {
+	async transform(value: AjaxPostBodyParams) {
+		if (value) {
```

Comment:
This transform pipe allows nest to validate the incoming request.
Since H5P does sent bodies with different shapes, this custom ValidationPipe makes sure the different cases are correctly validated.

Moving this between the controller and UC would mean the request would have to accept every input without proper validation

Reply:
Done created post.body.params.transform-pipe.ts